src/favicon_handler.rs

```diff
@@ -50,7 +50,7 @@ impl FaviconHandler {
     #[inline]
     pub fn is_favicon_request (req: &request::Request) -> bool {
         match req.origin.request_uri {
-            AbsolutePath(ref path) => path.as_slice() == "/favicon.ico",
+            AbsolutePath(ref path) => &path[0..] == "/favicon.ico",
```

Comment:
Alternatively `&*foo`, but perhaps slicing syntax is more explicit (I haven't checked the recent style guide on this)